mobile/library/cc/engine_builder.cc

```diff
@@ -150,6 +150,15 @@ EngineBuilder& EngineBuilder::addDnsPreresolveHostnames(const std::vector<std::s
   return *this;
 }
 
+EngineBuilder& EngineBuilder::setAdditionalSocketOptions(
+    const std::vector<envoy::config::core::v3::SocketOption>& socket_options) {
+  socket_options_.clear();
+  for (const auto& socket_option : socket_options) {
+    socket_options_.push_back(socket_option);
+  }
```

Comment:
Could we perhaps just do `socket_options_ = socket_options;`? Or maybe `socket_options_.clear();` should be removed?